test/pico_sha256_test/CMakeLists.txt

```
@@ -1,4 +1,5 @@
-if (NOT TARGET hardware_sha256)
+if (NOT TARGET pico_sha256)
+    message("Skipping hardware_sha256_test as pico_sha256 is unavailable on this platform")
```

I dunno if it'd be better for the message here to report the actual directory-name? (as it does for the other messages added in this PR)
```suggestion
    message("Skipping pico_sha256_test as pico_sha256 is unavailable on this platform")
```